File: Modules/Recommendations/Http/Resources/RecommendationResource.php
<?php

namespace Modules\Recommendations\Http\Resources;

use Illuminate\Http\Resources\Json\JsonResource;
use Modules\Lookups\Http\Resources\SubstanceCategoryResource;

class RecommendationResource extends JsonResource
{
    public function toArray($request): array
    {
        return [
            'id' => $this->id,
            'substance_category_id' => $this->substance_category_id,
            'type' => $this->type,
            'name_ar' => $this->name_ar,
            'name_en' => $this->name_en,
            'contact_info' => $this->contact_info,
            'latitude' => $this->latitude,
            'longitude' => $this->longitude,
            'is_active' => $this->is_active,
            'created_at' => $this->created_at,
            'updated_at' => $this->updated_at,
            'category' => new SubstanceCategoryResource($this->whenLoaded('category')),
        ];
    }
}

File: Solid.Api/Features/Recommendations/RecommendationResource.cs
using Solid.Api.Common;
using Solid.Api.Database.Entities;

namespace Solid.Api.Features.Recommendations;

public static class RecommendationResource
{
    public static object From(Recommendation recommendation)
    {
        return new
        {
            id = recommendation.Id,
            created_by_user_id = recommendation.CreatedByUserId,
            substance_category_id = recommendation.SubstanceCategoryId,
            type = recommendation.Type,
            name_ar = recommendation.NameAr,
            name_en = recommendation.NameEn,
            contact_info = recommendation.ContactInfo,
            latitude = recommendation.Latitude,
            longitude = recommendation.Longitude,
            is_active = recommendation.IsActive,
            created_at = EgyptDateTime.Format(recommendation.CreatedAt),
            updated_at = EgyptDateTime.Format(recommendation.UpdatedAt)
        };
    }
}
